src/routes/webhooks/paddle.ts

```diff
@@ -114,6 +118,24 @@ const getUserId = async ({
   return user.id;
 };
 
+const planChanged = async (data: SubscriptionUpdatedEvent) => {
+  const customData = data.data?.customData as { user_id: string };
+  const userId = await getUserId({
+    userId: customData?.user_id,
+    subscriptionId: 'subscriptionId' in data.data && data.data.subscriptionId,
+  });
+  const con = await createOrGetConnection();
+  const flags = await con.getRepository(User).findOne({
+    where: { id: userId },
+    select: ['subscriptionFlags'],
+  });
```

Comment:
We double query the user, once in `getUserId` then again in the call in this function?

Reply:
If we want to abstract anything maybe better that `getUserId` is just `getUser` and then we get user object either way.

Reply:
Yeah was thinking of that as well, just meant more refactors on end type as it's a optional return.

Reply:
Also the early return only cares about returning the id

Reply:
Alternative could be to have a seperate function for this case? (only 1 user now, but maybe future more)
Let me know if you prefer that

Reply:
I would just do it inline in both places but if we use util then I would go with `getUser`. Neither blocking for this PR.